### PR TITLE
Fix: hparams.yaml saved twice when using TensorBoardLogger

### LINKED ISSUE #5562: hparams.yaml saved twice when using TensorBoardLogger
## 🐛 Bug

When using `TensorBoardLogger(...default_hp_metric=False...)` and calling `self.save_hyperparameters()` from `LightningModule `instance, two _hparams.yaml_ files are created (unexpected one in the logger's `save_dir` and second under proper experiment version dir).

## Please reproduce using the BoringModel
See [**BoringModel**](https://colab.research.google.com/drive/1OpUpfU9awCEjPCR_XYo8z5TmF0MxBTJA?usp=sharing).

### To Reproduce

1. Create `LightningModule` with `self.save_hyperparameters()` call in `__init__`
2. Create `TensorBoardLogger` with `default_hp_metric=False`
3. Start training

### Expected behavior

Only one _hparams.yaml_ in `save_dir/name/version` should be created (like it happens when using tb logger with `default_hp_metric=True`).

### Environment
From Collab Notebook:

* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     True
	- pyTorch_version:   1.7.0+cu101
	- pytorch-lightning: 1.1.4
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.6.9
	- version:           1 SMP Thu Jul 23 08:00:38 PDT 2020

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Fixes #5562 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [ ] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — COMMENTED** [review]
can we get test for it?

**Comment by kaushikb11** [comment]
@Borda Done!

File: tests/loggers/test_tensorboard.py [hunk 1]
```diff
@@ -276,3 +276,14 @@ def test_tensorboard_finalize(summary_writer, tmpdir):
     logger.finalize("any")
     summary_writer().flush.assert_called()
     summary_writer().close.assert_called()
+
+
+def test_tensorboard_save_hparams_to_yaml(tmpdir):
+    model = EvalModelTemplate()
```
Comment (carmocca) [inline tests/loggers/test_tensorboard.py:282]: Can you use BoringModel instead?

Comment (kaushikb11) [reply tests/loggers/test_tensorboard.py]: @carmocca updated

**Review by carmocca — APPROVED** [review]

**Comment by rohitgr7** [comment]
we already the existence of `hparams.yaml` file here: https://github.com/PyTorchLightning/pytorch-lightning/blob/bdb6e1cd1134b4b2d46fc6a713059e0fbeb1b7d6/tests/loggers/test_tensorboard.py#L45-L50

can we somehow check whether it's created only once since that's the actual fix here?

**Comment by kaushikb11** [comment]
> we already the existence of `hparams.yaml` file here:
> 
> https://github.com/PyTorchLightning/pytorch-lightning/blob/bdb6e1cd1134b4b2d46fc6a713059e0fbeb1b7d6/tests/loggers/test_tensorboard.py#L45-L50
> 
> can we somehow check whether it's created only once since that's the actual fix here?

I don't see how the specific test for that would make any sense here.

**Comment by rohitgr7** [comment]
> > we already the existence of `hparams.yaml` file here:
> > https://github.com/PyTorchLightning/pytorch-lightning/blob/bdb6e1cd1134b4b2d46fc6a713059e0fbeb1b7d6/tests/loggers/test_tensorboard.py#L45-L50
> > 
> > can we somehow check whether it's created only once since that's the actual fix here?
> 
> I don't see how the specific test for that would make any sense here.

what I meant here is the test you added is somewhat already covered by one of the tests I linked above.

**Comment by kaushikb11** [comment]
@rohitgr7 Yeah, but the unit test is specific to `hparams.yaml` being saved. I don't see any issues having it.

**Comment by carmocca** [comment]
To warrant an addition, it needs to fail without the fix in the PR. Otherwise we are duplicating without actually testing the fix

**Comment by Borda** [comment]
@kaushikb11 thank you for sending your PR, just a minor issue coming from our side... we have swapped branches regarding upcoming feat 1.2, mind rebase on actual `master`?

**Pull request closed without merging**